### PR TITLE
Single files

### PR DESCRIPTION
**Pull request by gertjanal**
singlefiles() and delete()

### CODE REVIEW

File: src/modules/projectContext.js [hunk 1]
```diff
@@ -14,6 +14,15 @@ class ProjectContext {
         this.projectId = projectId;
     }
 
+    /**
+     * Delete the project.
+     *
+     * @returns A promise
+     */
+    delete = () => this.sessionManager.gatekeeper(`/projects/${this.projectId}`, {
```
Comment (ghost) [inline src/modules/projectContext.js]: The deletion for single file is `/singlefiles/{singleFileId}`

File: src/hansken.js [hunk 2]
```diff
@@ -19,6 +19,13 @@ class HanskenClient {
      */
     projects = () => this.sessionManager.gatekeeper('/projects').then((response) => response.json());
 
+    /**
+     * Get all single files.
+     *
+     * @returns All single files the current user is authorized for
+     */
+    singlefiles = () => this.sessionManager.gatekeeper('/projects?hidden=true').then((response) => response.json());
```
Comment (ghost) [inline src/hansken.js]: The api for this is `GET /singlefiles`

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/hansken.js [hunk 3]
```diff
@@ -19,13 +19,28 @@ class HanskenClient {
...
+    project = (projectId) => new ProjectContext(this.sessionManager, projectId, 'projects');
+
+    /**
+     * Get a context for a single file, to do project specific REST calls.
+     *
+     * @param {UUID} projectId The single file id
+     * @returns A ProjectContext for a single file
      */
-    project = (projectId) => new ProjectContext(this.sessionManager, projectId);
+     singlefile = (projectId) => new ProjectContext(this.sessionManager, projectId, 'singlefiles');
```
Comment (ghost) [inline src/hansken.js]: singlefileId

**Review by ghost — APPROVED** [review]

File: src/hansken.js [hunk 4]
```diff
@@ -19,13 +19,28 @@ class HanskenClient {
...
+    project = (projectId) => new ProjectContext(this.sessionManager, projectId, 'projects');
+
+    /**
+     * Get a context for a single file, to do project specific REST calls.
+     *
+     * @param {UUID} singlefileId The single file id
+     * @returns A ProjectContext for a single file
      */
-    project = (projectId) => new ProjectContext(this.sessionManager, projectId);
+     singlefile = (singlefileId) => new ProjectContext(this.sessionManager, singlefileId, 'singlefiles');
```
Comment (nfi-jd) [inline src/hansken.js]:
```suggestion
    singlefile = (singlefileId) => new ProjectContext(this.sessionManager, singlefileId, 'singlefiles');
```

File: src/modules/projectContext.js [hunk 5]
```diff
@@ -7,55 +7,66 @@ class ProjectContext {
...
      *
      * @param {UUID} The image id
      * @returns A new ProjectImageContext
      */
-    image = (imageId) => new ProjectImageContext(this.sessionManager, this.projectId, imageId);
+    image = (imageId) => new ProjectImageContext(this.sessionManager, this.id, imageId);
 
     /**
-     * Create a ProjectSearchContext to search for traces, facets, tracelets and more.
+     * Create a SearchContext to search for traces, facets, tracelets and more.
```
Comment (nfi-jd) [inline src/modules/projectContext.js]:
```suggestion
     * Create a ProjectSearchContext to search for traces, facets, tracelets and more.
```

File: src/modules/projectContext.js [hunk 6]
```diff
@@ -7,55 +7,66 @@ class ProjectContext {
      * Create a context for a specific project. This can be used to search in a project or list its images.
      *
      * @param {SessionManager} sessionManager The session manager, used for connections to the Hansken servers
-     * @param {UUID} projectId The project id
+     * @param {UUID} id The project id or single file id
+     * @param {String} collection 'projects' or 'singlefiles'
```
Comment (nfi-jd) [inline src/modules/projectContext.js]:
```suggestion
     * @param {'projects' | 'singlefiles'} collection The name of the collection to use
```

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "hansken-js",
-    "version": "0.3.0",
+    "version": "0.3.1",
```
Comment (nfi-jd) [inline package.json]: New functionality ➛ bump minor version (https://semver.org/)
```suggestion
    "version": "0.4.0",
```

**Review by nfi-jd — CHANGES_REQUESTED** [review]

**Review by nfi-jd — APPROVED** [review]

**Pull request merged**